x-pack/plugins/endpoint/public/applications/endpoint/store/index.ts

```diff
@@ -4,44 +4,25 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import {
-  createStore,
-  compose,
-  applyMiddleware,
-  Store,
-  MiddlewareAPI,
-  Dispatch,
-  Middleware,
-} from 'redux';
+import { createStore, compose, applyMiddleware, Store } from 'redux';
 import { CoreStart } from 'kibana/public';
 import { appReducer } from './reducer';
 import { alertMiddlewareFactory } from './alerts/middleware';
 import { hostMiddlewareFactory } from './hosts';
 import { policyListMiddlewareFactory } from './policy_list';
 import { policyDetailsMiddlewareFactory } from './policy_details';
-import { GlobalState, MiddlewareFactory } from '../types';
-import { AppAction } from './action';
+import { ImmutableMiddlewareFactory, SubstateMiddlewareFactory } from '../types';
 import { EndpointPluginStartDependencies } from '../../../plugin';
 
 const composeWithReduxDevTools = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
   ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({ name: 'EndpointApp' })
   : compose;
 
-export type Selector<S, R> = (state: S) => R;
```

Comment:
moved type to `types` module. Code should be more readable now, and this type is relevant to the others there: `ImmutableMiddleware`, `ImmutableMiddlewareAPI`, `ImmutableMidlewareFactory`.